atomix/cluster/src/test/java/io/atomix/raft/RaftFailOverTest.java

```diff
@@ -118,19 +117,97 @@ public void shouldRecoverLeaderRestart() throws Throwable {
   }
 
   @Test
-  @Ignore("https://github.com/zeebe-io/zeebe/issues/4467")
-  public void testNodeCatchUpAfterCompaction() throws Exception {
+  public void shouldDoSnapshot() throws Exception {
     // given
-    raftRule.shutdownServer("1");
+    raftRule.appendEntries(128);
+
+    // when
+    raftRule.doSnapshot(100);
+
+    // then
+    assertThat(raftRule.allNodesHaveSnapshotWithIndex(100)).isTrue();
+  }
+
+  @Test
+  public void shouldCompactLogOnSnapshot() throws Exception {
+    // given
+    raftRule.appendEntries(128);
+    final var memberLogs = raftRule.getMemberLogs();
+
+    // when
+    raftRule.doSnapshot(100);
+
+    // then
+    final var compactedLogs = raftRule.getMemberLogs();
+
+    assertThat(compactedLogs.isEmpty()).isFalse();
+    for (final String raftMember : compactedLogs.keySet()) {
+      final var compactedLog = compactedLogs.get(raftMember);
+      final var previousLog = memberLogs.get(raftMember);
+      assertThat(compactedLog.size()).isLessThan(previousLog.size());
+      assertThat(compactedLog).isSubsetOf(previousLog);
+    }
+  }
+
+  @Test
+  public void shouldReplicateSnapshotOnJoin() throws Exception {
+    // given
+    final var follower = raftRule.shutdownFollower();
+    raftRule.appendEntries(128);
+    raftRule.doSnapshot(100);
+
+    // when
+    raftRule.startNode(follower);
+
+    // then
+    assertThat(raftRule.allNodesHaveSnapshotWithIndex(100)).isTrue();
+    final var snapshot = raftRule.snapshotOnNode(follower);
+
+    assertThat(snapshot.index()).isEqualTo(100);
+    assertThat(snapshot.term()).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldTakeMultipleSnapshotAndReplicateSnapshotAfterRestart() throws Exception {
+    // given
+    raftRule.appendEntries(128);
+    raftRule.doSnapshot(100);
+    final var follower = raftRule.shutdownFollower();
+    raftRule.appendEntries(128);
+    raftRule.doSnapshot(200);
+    raftRule.appendEntries(128);
+    raftRule.doSnapshot(300);
+
+    // when
+    raftRule.startNode(follower);
```

Comment:
In addition to this, another test that would be interesting is to restart follower with data loss. Can you add it?

Reply:
Of course!

Reply:
btw this should be similar to shouldReplicateSnapshotOnJoin

Reply:
It is different. When follower restarts with data loss, leader thinks that it has already send those events to the follower. There was a bug before - something like leader doesn't send the snapshot again if the follower restarts with data loss.